src/H5Pdxpl.c

```diff
@@ -2386,3 +2496,248 @@ H5Pset_dataset_io_hyperslab_selection(hid_t plist_id, unsigned rank, H5S_seloper
 
     FUNC_LEAVE_API(ret_value)
 } /* end H5Pset_dataset_io_hyperslab_selection() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5Pset_selection_io
+ *
+ * Purpose:     To set the selection I/O mode in the dataset
+ *              transfer property list.
+ *
+ * Note:        The library may not perform selection I/O as it asks for if
+ *              the layout callback determines that it is not feasible to do so.
+ *
+ * Return:      Success:    Non-negative
+ *              Failure:    Negative
+ *
+ * Programmer:  Vailin Choi
+ *              March 5, 2023
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5Pset_selection_io(hid_t plist_id, H5D_selection_io_mode_t selection_io_mode)
+{
+    H5P_genplist_t *plist;               /* Property list pointer */
+    herr_t          ret_value = SUCCEED; /* Return value */
+
+    FUNC_ENTER_API(FAIL)
+    H5TRACE2("e", "iDC", plist_id, selection_io_mode);
+
+    /* Check arguments */
+    if (plist_id == H5P_DEFAULT)
+        HGOTO_ERROR(H5E_PLIST, H5E_BADVALUE, FAIL, "can't set values in default property list")
+
+    if (NULL == (plist = H5P_object_verify(plist_id, H5P_DATASET_XFER)))
+        HGOTO_ERROR(H5E_PLIST, H5E_BADTYPE, FAIL, "not a dxpl")
+
+    /* Set the selection I/O mode */
+    if (H5P_set(plist, H5D_XFER_SELECTION_IO_MODE_NAME, &selection_io_mode) < 0)
+        HGOTO_ERROR(H5E_PLIST, H5E_CANTSET, FAIL, "unable to set value")
+
+done:
+    FUNC_LEAVE_API(ret_value)
+} /* end H5Pset_selection_io() */
+
+/*-------------------------------------------------------------------------
+ * Function:    H5Pget_selection_io
+ *
+ * Purpose:     To retrieve the selection I/O mode that is set in
+ *              the dataset transfer property list.
+ *
+ * Note:        The library may not perform selection I/O as it asks for if
+ *              the layout callback determines that it is not feasible to do so.
+ *
+ * Return:      Success:    Non-negative
+ *              Failure:    Negative
+ *
+ * Programmer:  Vailin Choi
+ *              March 5, 2023
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5Pget_selection_io(hid_t plist_id, H5D_selection_io_mode_t *selection_io_mode /*out*/)
```

Comment:
Should check for H5P_DEFAULT and set plist_id to correct library default DXPL if that's passed in

Reply:
Are you sure about this? None of the other H5Pget* functions seem to do this

Reply:
I guess it might be useful to make this work, especially since we're bad about documenting the default values for properties (I just had to scour through the code to find some to fix one of the above tests), but we should really apply it to all at once as a separate PR

Reply:
Yeah I think we should check for and handle H5P_DEFAULT in every API routine where a plist is passed in. I think it's just something that's easy to forget and so many routines are missing the check. Maybe we should move the logic down into H5P_object_verify since it accepts a "plist type" parameter. Other places where H5I_object is used won't be covered but it'd be a step in the right direction.